### PR TITLE
Ruby code sample

### PR DESCRIPTION
**Pull request by tomelm**
Provides a Ruby code sample.

### CODE REVIEW

**Comment by tomelm** [comment]
@mittonk @ajm188 giving this to both of you.

@Yelp/partnerships have at it.

File: fusion/ruby/sample.rb [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+# Place holders for Yelp Fusion's OAuth 2.0 credentials. Grab them
+# from https://www.yelp.com/developers/v3/manage_app
+CLIENT_ID = nil
+CLIENT_SECRET = nil
+
+
+# Constants, do not change these
+API_HOST = "https://api.yelp.com"
+SEARCH_PATH = "/v3/businesses/search"
+BUSINESS_PATH = "/v3/businesses/"  # trailing / because we append the business id to the path
```
Comment (mittonk) [inline fusion/ruby/sample.rb]: Nicely clear, thanks.

File: fusion/ruby/sample.rb [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+TOKEN_PATH = "/oauth2/token"
+GRANT_TYPE = "client_credentials"
+
+
+DEFAULT_BUSINESS_ID = "yelp-san-francisco"
+DEFAULT_TERM = "dinner"
+DEFAULT_LOCATION = "San Francisco, CA"
+SEARCH_LIMIT = 5
```
Comment (mittonk) [inline fusion/ruby/sample.rb]: We often use 3 for examples, easier to read on a screen.  I'm fine with 5.

File: fusion/ruby/sample.rb [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+#
+#   bearer_token
+#   # => "Bearer some_fake_access_token"
+#
+# Returns your access token
+def bearer_token
+  # Put the url together
+  url = "#{API_HOST}#{TOKEN_PATH}"
+
+  raise "Please set your CLIENT_ID" if CLIENT_ID.nil?
```
Comment (mittonk) [inline fusion/ruby/sample.rb]: Good, I definitely think this assertion is worth while here.

File: fusion/ruby/sample.rb [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+#   business("yelp-san-francisco")
+#   # => {
+#          "name": "Yelp",
+#          "id": "yelp-san-francisco"
+#          ...
+#        }
+#
+# Returns a parsed json object of the request
+def business(business_id)
+  url = "#{API_HOST}#{BUSINESS_PATH}#{business_id}"
```
Comment (mittonk) [inline fusion/ruby/sample.rb]: Do we need
`ERB::Util.url_encode(business_id)`
like in still-open pull request https://github.com/Yelp/yelp-ruby/pull/55 ?

File: fusion/ruby/sample.rb [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+
+  response = search(term, location)
+
+  puts "Found #{response["total"]} businesses. Listing #{SEARCH_LIMIT}:"
+  response["businesses"].each {|biz| puts biz["name"]}
+when "lookup"
+  business_id = options.fetch(:id, DEFAULT_BUSINESS_ID)
+
+
+  raise "term is not a valid parameter for lookup" if options.key?(:term)
```
Comment (mittonk) [inline fusion/ruby/sample.rb]: Hm, I think term may actually be optional.  Don't care much.

File: fusion/ruby/sample.rb (hunk 4 above)
Comment (tomelm) [inline fusion/ruby/sample.rb]: It looks like it's working for me: https://gist.github.com/tomelm/7b257c4261e029f4afcfc0e83fce1ee0

File: fusion/ruby/sample.rb (hunk 5 above)
Comment (tomelm) [inline fusion/ruby/sample.rb]: It is on the API, I was making it required here for the demo

File: fusion/ruby/sample.rb (hunk 4 above)
Comment (mittonk) [inline fusion/ruby/sample.rb]: Sounds good then.

File: fusion/ruby/sample.rb (hunk 5 above)
Comment (mittonk) [inline fusion/ruby/sample.rb]: Fair enough.

**Pull request merged**